tests/framework/extensions/nodes/ec2/ec2.go

```diff
@@ -10,138 +10,174 @@ import (
+			InstanceIds: listOfInstanceIds,
+		})
+		if err != nil {
+			return nil, nil, err
+		}
 
-	// describe instance to get attributes=
-	describe, err := ec2Client.SVC.DescribeInstances(&ec2.DescribeInstancesInput{
-		InstanceIds: listOfInstanceIds,
-	})
-	if err != nil {
-		return nil, err
-	}
+		// describe instance to get attributes=
+		describe, err := ec2Client.SVC.DescribeInstances(&ec2.DescribeInstancesInput{
+			InstanceIds: listOfInstanceIds,
+		})
+		if err != nil {
+			return nil, nil, err
+		}
 
-	readyInstances := describe.Reservations[0].Instances
+		readyInstances := describe.Reservations[0].Instances
 
-	sshKey, err := nodes.GetSSHKey(ec2Client.Config.AWSSSHKeyName)
-	if err != nil {
-		return nil, err
-	}
+		sshKey, err := nodes.GetSSHKey(config.AWSSSHKeyName)
+		if err != nil {
+			return nil, nil, err
+		}
 
-	var ec2Nodes []*nodes.Node
+		for _, readyInstance := range readyInstances {
+
+			if multiconfig {
+				if config.AWSAMI == ec2Client.ClientConfig.AWSEC2Config[0].AWSAMI &&
+					config.AWSUser == ec2Client.ClientConfig.AWSEC2Config[0].AWSUser {
+					ec2Node := &nodes.Node{
+						NodeID:          *readyInstance.InstanceId,
+						PublicIPAddress: *readyInstance.PublicIpAddress,
+						SSHUser:         config.AWSUser,
+						SSHKey:          sshKey,
+					}
+					ec2Nodes = append(ec2Nodes, ec2Node)
+				}
+
+				if config.AWSAMI == ec2Client.ClientConfig.AWSEC2Config[1].AWSAMI &&
+					config.AWSUser == ec2Client.ClientConfig.AWSEC2Config[1].AWSUser {
+					ec2Node2 := &nodes.Node{
+						NodeID:          *readyInstance.InstanceId,
+						PublicIPAddress: *readyInstance.PublicIpAddress,
+						SSHUser:         config.AWSUser,
+						SSHKey:          sshKey,
+					}
+					ec2Nodes2 = append(ec2Nodes2, ec2Node2)
+				}
+			}
+			if !multiconfig {
+				ec2Node := &nodes.Node{
+					NodeID:          *readyInstance.InstanceId,
+					PublicIPAddress: *readyInstance.PublicIpAddress,
+					SSHUser:         config.AWSUser,
+					SSHKey:          sshKey,
+				}
+				ec2Nodes = append(ec2Nodes, ec2Node)
+				ec2Nodes2 = nil
+			}
+		}
+
+		client.Session.RegisterCleanupFunc(func() error {
+			return DeleteNodes(client, ec2Nodes, ec2Nodes2, multiconfig)
+		})
 
-	for _, readyInstance := range readyInstances {
-		ec2Node := &nodes.Node{
-			NodeID:          *readyInstance.InstanceId,
-			PublicIPAddress: *readyInstance.PublicIpAddress,
-			SSHUser:         ec2Client.Config.AWSUser,
-			SSHKey:          sshKey,
+		if !multiconfig {
+			break
 		}
-		ec2Nodes = append(ec2Nodes, ec2Node)
 	}
 
-	client.Session.RegisterCleanupFunc(func() error {
-		return DeleteNodes(client, ec2Nodes)
-	})
-
-	return ec2Nodes, nil
+	return ec2Nodes, ec2Nodes2, nil
 }
 
 // DeleteNodes terminates ec2 instances that have been created.
-func DeleteNodes(client *rancher.Client, nodes []*nodes.Node) error {
+func DeleteNodes(client *rancher.Client, nodes []*nodes.Node, nodes2 []*nodes.Node, multiconfig bool) error {
```

Comment:
another nit, don't like the naming for nodes2

Reply:
What would you prefer it to be?

Reply:
Changed to `winEC2Nodes`.